indico/modules/rb_new/util.py

```diff
@@ -638,19 +640,35 @@ def get_room_blockings(start_dt=None, end_dt=None, created_by=None, in_rooms_own
     return query.all()
 
 
-def create_blocking(rooms, start_date, end_date, reason, allowed_principals):
+def _populate_blocking(blocking, room_ids, reason, allowed_principals, start_dt=None, end_dt=None):
+    blocking_data = {'reason': reason, 'created_by_user': session.user,
+                     'allowed': [GroupProxy(pr['id'], provider=pr['provider'])
+                                 if pr.get('is_group')
+                                 else User.query.filter(User.id == pr['id']).one()
+                                 for pr in allowed_principals],
+                     'blocked_rooms': [BlockedRoom(room_id=room.id)
+                                       for room in Room.query.filter(Room.id.in_(room_ids))]}
+
+    if start_dt:
+        blocking_data['start_date'] = start_dt
+    if end_dt:
+        blocking_data['end_date'] = end_dt
+    blocking.populate_from_dict(blocking_data)
```

Comment:
Why do we build a dict and populate from that instead of simply assigning things directly?